lib/ui/views/course_landing/course_landing_view.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/material.dart';
+import 'package:responsive_builder/responsive_builder.dart';
+import 'package:stacked/stacked.dart';
+
+import 'course_landing_view.desktop.dart';
+import 'course_landing_view.mobile.dart';
+import 'course_landing_viewmodel.dart';
+
+class CourseLandingView extends StackedView<CourseLandingViewModel> {
+  const CourseLandingView({super.key});
+
+  @override
+  Widget builder(
+    BuildContext context,
+    CourseLandingViewModel viewModel,
+    Widget? child,
+  ) {
+    return ScreenTypeLayout.builder(
+      mobile: (_) => const CourseLandingViewMobile(),
+      desktop: (_) => const CourseLandingViewDesktop(),
+    );
+  }
+
+  @override
+  CourseLandingViewModel viewModelBuilder(
+    BuildContext context,
+  ) =>
+      CourseLandingViewModel(courseId: 'flutter-web');
```

Comment:
We actually want the path of this to include this id and get it from there. We should discuss this when you're up but what I'm currently thinking is:

- `course/flutter-web` shows `CourseLandingView`
- `course/flutter-web/chapterId` shows `CourseDetailsView` with the chapter selected. 

This will require some updates, which are luckily now all detailed in the [new navigation video ](https://youtu.be/cGdNRnBdWpw)

Reply:
I don't know if there is a bug on the `RouterService` or I'm doing something wrong but when I navigate to `CourseDetailsView` with `chapterId`, the chapterId is not taken into account because is null and the first chapterId is always chosen.

Also, I'm not sure how we are managing the routing here because for what I think, CourseLandingView and CourseDetailsView should be the same. Why the content of CourseLandingView is not on CourseDetailsView?

With the current setup, we must pass CourseLandingView first to the Stacked routes, if not it is ignored because CourseDetailsView route would match first.

Let's talk about this tomorrow. I believe our routes should be like below:

`/courses/:courseId/chapters/:chapterId`

being resources always in plural. If you see any other site, the convention is always like this.

Reply:
@ferrarafer 

- `CourseLandingView` is the view that shows the course overview and the course trailer. 
- `CorseDetailsView` is the view where they consume the content
- `CourseChapterView` is nested in the `CourseDetailsView`

we'll discuss later and you can take me through it and why it's not working. I agree on the plural for the path, but I don't think we need the `/chapter` since we're not really navigating to chapters. But we'll discuss soon.